jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/SyncServerlessAppAction.kt

```diff
@@ -110,6 +116,12 @@ class SyncServerlessAppAction : AnAction(
                         }
                     )
                 )
+                SamTelemetry.sync(
+                    project = project,
+                    result = Result.Failed,
+                    syncedResources = SyncedResources.AllResources,
+                    reason = "Older SAM CLI version"
```

Comment:
are these messages standardized?

Reply:
They're not standardized but I suggest an enum-like naming. Here are some examples from vscode toolkit (`code` is used to fill the telemetry `reason`):

```
src/lambda/commands/createNewSamApp.ts|161 col 13| reason = 'userCancelled'
src/lambda/commands/createNewSamApp.ts|204 col 13| reason = 'fileNotFound'
src/lambda/commands/createNewSamApp.ts|295 col 13| reason = 'complete'
src/lambda/commands/createNewSamApp.ts|298 col 13| reason = 'fileNotFound'
src/auth/credentials/sharedCredentials.ts|245 col 13| code: 'ProfileAlreadyExists',
src/auth/credentials/validation.ts|118 col 13| code: 'InvalidCredentialsData',
src/auth/sso/model.ts|101 col 17| code: trustedDomainCancellation,
src/auth/sso/ssoAccessTokenProvider.ts|229 col 13| code: 'TimedOut',
src/auth/ui/vue/show.ts|224 col 20| (e.code === trustedDomainCancellation || e.cause?.name === trustedDomainCancellation)
src/auth/utils.ts|291 col 13| code: 'InvalidCredentials',
src/auth/utils.ts|306 col 17| code: 'MissingConnection',
```

Reply:
Thanks for this, i'll update the reason fields here to have enum-like values.